src/figuresClasses.ts

```diff
@@ -1,19 +1,85 @@
-export interface Figure {
+// import { CpuInfo } from "os";
 
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle'
 }
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle {
-
+export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea():number,
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape: Shape = Shape.Triangle;
 
+  constructor(
+    public color: Color,
+    private a: number,
+    private b: number,
+    private c: number,
+  ) {
+    if (
+      this.a + this.b <= this.c
+      || this.b + this.c <= this.a
+      || this.c + this.a <= this.b
+    ) {
+      throw new Error('This is not a triangle');
+    }
+  }
+
+  getArea(): number {
+    const semiPerimeter = (this.a + this.b + this.c) / 2;
+    const square = Math.sqrt(
+      semiPerimeter
+      * (semiPerimeter - this.a)
+      * (semiPerimeter - this.b)
+      * (semiPerimeter - this.c),
+    );
+
+    return +square.toFixed(2);
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape: Shape = Shape.Circle;
+
+  constructor(public color: Color, private radius: number) {
+    if (this.radius <= 0) {
+      throw new Error('this is not a circle');
+    }
+  }
 
+  getArea(): number {
+    const square = Math.PI * this.radius ** 2;
+
+    return Math.floor(square * 100) / 100;
+  }
 }
 
-export function getInfo(figure) {
+export class Rectangle implements Figure {
+  shape: Shape = Shape.Rectangle;
+
+  constructor(
+    public color: Color,
+    private width: number,
+    private height: number,
+  ) {
+    if (this.width <= 0 || this.height <= 0) {
+      throw new Error('this is not a Rectangle');
+    }
```

Comment:
same here

Reply:
you can create separate method, that will accept input arguments and will check that all of them is greater than 0
and you can use this function for checking input arguments for each constructor